prime-router/src/main/kotlin/cli/helpers/HL7DiffHelper.kt

```diff
@@ -0,0 +1,334 @@
+package gov.cdc.prime.router.cli.helpers
+
+import ca.uhn.hl7v2.HL7Exception
+import ca.uhn.hl7v2.model.Composite
+import ca.uhn.hl7v2.model.Group
+import ca.uhn.hl7v2.model.Message
+import ca.uhn.hl7v2.model.Primitive
+import ca.uhn.hl7v2.model.Segment
+import ca.uhn.hl7v2.model.Structure
+import ca.uhn.hl7v2.model.Type
+import ca.uhn.hl7v2.model.Varies
+import org.apache.commons.lang3.StringUtils
+
+class HL7DiffHelper {
+
+    fun filterNames(message: Message, names: Array<String>, map: MutableMap<String, Segment>) {
+        names.filter { name -> message.getAll(name).isNotEmpty() }.forEach { messageName ->
+            val children = message.getAll(messageName)
+            children.forEachIndexed { index, c ->
+                indexStructure(c, (index + 1).toString(), map)
+            }
+        }
+    }
+
+    /**
+     * Does the diffing of the [input] message to the [output] message. Results are echoed on the command line.
```

Comment:
Does this actually echo results to command line? It probably shouldn't? The reporting of the functionality should be split out from the actually business logic -- which I think it is? So remove this part of the comment?